data-config-tabs.json

```diff
@@ -0,0 +1,87 @@
+{
```

Comment:
having this as a completely separate, mostly duplicate config file is confusing and likely to cause example config files to get out of sync.  Since this is simply adding a new, optional feature, let's make it work with the existing config file.

Tabs should be something that is supported as an option inside any of "properties", "properties-shared", and "properties-global", all individually, based on what the user puts into the config.  Perhaps it's as simple as supporting a "tab" value for "type" of field that triggers the code to create a new tab and include the fields within.

Reply:
Agreed, I want to make sure that the initial set is validated before merging into the main file. I will update the main config file instead of creating a separate one.

Reply:
New pattern looks great - all my current review comments are small and specific coding items.  Overall I'm very excited about this addition, and the new implementation seems to be much less code than the original PR submission.  Good work @sudheerdvn!